core/mpi/controller/paciente.ts

```diff
@@ -217,21 +217,17 @@ export async function matching(data) {
                 // };
                 const words = data.cadenaInput.split(' ');
                 let andQuery = [];
-
-                words.forEach(w => {
-                    let regexp = new RegExp(w, 'i');
+                const regexToken = words.map((token) => { return new RegExp(`^.*${token.toLowerCase()}.*$`, 'i'); });
+                query = { tokens: { $in: regexToken } };
```

Comment:
El operador $in trabaja internamente como un $or. Osea que si busco 2 _tokens_ _federico perez_ me va a traer todos los Perez posibles aunque no se llamen Federico